### DISCUSSION TITLE
Spacing transform for 2D data

### DISCUSSION THREAD

**Discussion by AdamKlasa** [Project-MONAI/MONAI#4265] [Q&A]
I'm working on 2D ultrasound segmentation/regression. The dataset is a proprietary collection of images that I have both in DICOM as well as png formats with binary masks as labels. The size of images vary, as well as their pixel spacing. My idea is to train a UNet based model that would also be capable of performing regression and outputting predicted measurements on top of segmentations. I thought it would be a good idea to use the spacing transform to ensure pixel spacing is identical in all of the images. 
Does this functionality even work for 2D data "out-of-the-box"? Shape of the data is 1xHxW (single channel). My attempts so far result in a generic error ("RuntimeError: applying transform <monai.transforms.spatial.array.Spacing object at 0x7f407a6f2c40>") . What is the proper order of transformations? For now I'm trying to keep it simple:
I'm using monai version 0.8.1.

```
my_transforms = Compose([
    LoadImage(image_only=True),
    ToTensor(),
    Spacing(1),
])
```

**Comment by wyli** [comment]
Hi @AdamKlasa the transforms can handle 2D, but with the images in png format and/or with `image_only=True` the spacing transform won't have access to the original spacing information. There are examples of handling this by customizing the transforms, for example https://github.com/Project-MONAI/tutorials/blob/master/modules/image_dataset.ipynb another approach would be saving the masks into nifti files which has spacing info, then follow this example https://github.com/Project-MONAI/tutorials/blob/a329f5fe3c09162dbd0966dfc93582de75d7b748/2d_segmentation/torch/unet_training_dict.py#L59-L64
